content/posts/observability-in-the-cloud.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Observability in the Cloud
+author: Gabriel Koch
+comments: true
+date: 2022-04-01
+---
+
+This post aims to explain how observability and monitoring should be distinguished, what the problems with "traditional" monitoring is when we deal with distributed cloud systems and how observability can alleviate these problems.
+Finally I roughly outline how it should be approached.
+
+## Observability vs. Monitoring
+
+For the difference between monitoring and observability, I recommend Cindy Sridharan's book Distributed Systems Observability and Charity Major's various blog posts on the topic (e.g. [Observability is a Many-Splendored Definition](https://charity.wtf/2020/03/03/observability-is-a-many-splendored-thing/)) who go into more detail.
+
+Here, I summarize the main points and draw some conclusions.
+
+Monitoring is what most of us are used to.
+It encompasses collection of telemetry (metrics, logs and traces), visualization through dashboards and alerting, with metrics being the primary data type used.
+Monitoring is primarily the process of collecting telemetry and acting upon it.
+It deals with the known-unknowns (e.g. we know that we may run out of memory or disk and can monitor it, but we do not precisely know when and because of which processes it happens).
```

Comment:
```suggestion
It deals with the known-unknowns (e.g. we know that we may run out of memory or disk and we can monitor it, but we do not precisely know when it will happen and which processes causes it to happen).
```